clarifai/client/search.py

```diff
@@ -228,20 +236,18 @@ def query(self, ranks=[{}], filters=[{}]):
           user_app_id=self.user_app_id,
           searches=[
               resources_pb2.Search(
-                  query=resources_pb2.Query(filters=all_filters), metric=self.metric_distance)
+                  query=resources_pb2.Query(ranks=all_ranks, filters=all_filters),
+                  metric=self.metric_distance)
           ])
 
       return self.list_all_pages_generator(self.STUB.PostInputsSearches,
                                            service_pb2.PostInputsSearchesRequest, request_data)
 
     # Calls PostAnnotationsSearches for annotation ranks, filters
-    rank_annot_proto, filters_annot_proto = [], []
-    for rank_dict in ranks:
-      rank_annot_proto.append(self._get_annot_proto(**rank_dict))
```

Comment:
How come we do not allow ranking by annotation anymore?

Reply:
It is supported, I've shifted it to L221:225

Reply:
Ah my bad, I see it now.